contracts/mainnet/connectors/compound/v3/main.sol

```diff
@@ -0,0 +1,845 @@
+	 * @param to The address on behalf of which the borrow is to be repaid.
+	 * @param amt The amount to be repaid.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens repaid.
+	 */
+	function paybackOnBehalf(
+		address market,
+		address token,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+		require(
+			market != address(0) && token != address(0) && to != address(0),
+			"invalid market/token/to address"
+		);
+
+		address token_ = getBaseToken(market);
+		bool isEth = token == ethAddr;
+		require(token == token_ || token == ethAddr, "invalid-token");
+
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		amt_ = amt_ == uint256(-1)
+			? CometInterface(market).borrowBalanceOf(to)
+			: amt_;
+
+		uint256 borrowBal = CometInterface(market).borrowBalanceOf(to);
+		if (borrowBal > 0) {
+			require(amt_ <= borrowBal, "repay-amt-greater-than-debt");
+		}
+
+		convertEthToWeth(isEth, tokenContract, amt_);
+		approve(tokenContract, market, amt_);
+
+		CometInterface(market).supplyTo(to, token_, amt_);
+
+		setUint(setId, amt_);
+
+		eventName_ = "LogPaybackOnBehalf(address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token_, to, amt_, getId, setId);
+	}
+
+	/**
+	 * @dev Repays entire borrow of the base asset form 'from' on behalf of 'to'.
+	 * @notice Repays an entire borrow of the base asset on behalf of 'to'. Approve the comet markey
+	 * @param market The address of the market.
+	 * @param token The address of the token to be repaid. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param from The address from which the borrow has to be repaid on behalf of 'to'.
+	 * @param to The address on behalf of which the borrow is to be repaid.
+	 * @param amt The amount to be repaid.
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens repaid.
+	 */
+	function paybackFromUsingManager(
+		address market,
+		address token,
+		address from,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		external
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+		require(
+			market != address(0) && token != address(0) && to != address(0),
+			"invalid market/token/to address"
+		);
+
+		address token_ = getBaseToken(market);
+		bool isEth = token == ethAddr;
+		require(token == token_ || token == ethAddr, "invalid-token");
+
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		amt_ = setAmt(market, token_, from, amt_, isEth, ACTION.repay);
+
+		uint256 borrowBal = CometInterface(market).borrowBalanceOf(to);
+		if (borrowBal > 0) {
+			require(amt_ <= borrowBal, "repay-amt-greater-than-debt");
+		}
+
+		approve(tokenContract, market, amt_);
```

Comment:
@Richa-iitr What if `from` is different address?

Reply:
@Richa-iitr I think this function is already in the `setAmt` function

Reply:
This needs to removed.